src/main/java/cart/domain/MemberEntity.java

```diff
@@ -0,0 +1,30 @@
+package cart.domain;
+
+public class MemberEntity {
+
+    private final Long id;
+    private final Email email;
+    private final Password password;
+
+    public MemberEntity(final Long id, final String address, final String password) {
+        this.id = id;
+        this.email = new Email(address);
+        this.password = new Password(password);
+    }
+
```

Comment:
값객체 사용 💯
애초에 MemberEntity에서 Email, Password를 받는건 어떤가요?

```suggestion

    private final Long id;
    private final Email email;
    private final Password password;

    public MemberEntity(final Long id, final Email email, final Password password) {
        this.id = id;
        this.email = email;
        this.password = password;
    }

```

Reply:
감사합니다! Email과 Password를 객체로 적절히 만든 것 같아 뿌듯하네요!

제가 MemberEntity에게 Email과 Password 생성 책임을 둔 이유는 다음과 같습니다.

- 회원이 생성될 때, Email과 Password가 함께 생성되는 것이 맞다고 생각했습니다. 회원이 존재하는 경우에만, 그 회원의 Email과 Password가 존재하는 것이 합당하다고 느껴졌기 때문입니다. 만약 MemberEntity 밖에서 Email과 Password가 생성된다면, 회원은 존재하지 않는데, Email과 Password가 생성된 경우가 있게 됩니다. 이것이 각 객체의 맥락에 어긋나게 느껴졌습니다.
- GRASP 의 Creator 원칙이 이 상황에 맞다고 생각했습니다. Email과 Password는 MemberEntity 객체의 일부입니다. 따라서 MemberEntity 객체에게 생성 책임을 부여했습니다.

아직 미숙해서 시야가 좁습니다. 혹시 MemberEntity에서 Email, Password를 받는 것을 권유해주신 특별한 이유가 있으실까요? 알려주신다면 공부하고 고민해보겠습니다!! 😀